Implement RFC 209: Extension discoverability

tjkandala opened the pull request:
Issue #13329

AlicjaSuska commented:
Hi, @tjkandala thank you for sharing this PR, here are things that I've noticed:

---
- The extension name should be 'H4' not 'a' only. It will change the color from link blue to header grey-blue
<img width="130" alt="Screenshot 2020-09-10 at 14 26 17" src="https://user-images.githubusercontent.com/20326070/92728958-e4944e00-f371-11ea-849b-b38416c007b8.png">

---

- WIP badge should be using $warning (orange) not $danger (red)
<img width="46" alt="Screenshot 2020-09-10 at 14 28 19" src="https://user-images.githubusercontent.com/20326070/92729013-fa097800-f371-11ea-9da2-cc4648d1d96a.png">

---

- 'Other' should be the last category in the filters
<img width="118" alt="Screenshot 2020-09-10 at 14 29 26" src="https://user-images.githubusercontent.com/20326070/92729076-10afcf00-f372-11ea-9a14-abcb74c9ec82.png">

---

- puzzle illustration's opacity is too low (both light and dark themes)
<img width="119" alt="Screenshot 2020-09-10 at 14 29 51" src="https://user-images.githubusercontent.com/20326070/92729112-20c7ae80-f372-11ea-9330-f798c274b863.png">

---

- CTA banner body text should have regular body color, this one is very light (light theme only)
<img width="622" alt="Screenshot 2020-09-10 at 14 30 41" src="https://user-images.githubusercontent.com/20326070/92729208-4b196c00-f372-11ea-9b75-fb665c012db5.png">

---

- extension name in the grey alert (when disabling) should be semi-bold, not bold. Also, for the dark theme, it still has a link color instead of the alert text color.
<img width="437" alt="Screenshot 2020-09-10 at 14 32 53" src="https://user-images.githubusercontent.com/20326070/92729498-b3684d80-f372-11ea-849b-bd0b32f59f53.png">
<img width="192" alt="Screenshot 2020-09-10 at 14 32 37" src="https://user-images.githubusercontent.com/20326070/92729503-b6633e00-f372-11ea-81e4-fc35bd0765c0.png">

---

- when enabling the extension, alert background color should have 20% opacity for both light and dark themes ([Figma](https://www.figma.com/file/7Ek89vyDoqlGhCn3wyS40J/11872-Extensions-discoverability-Final-review?node-id=480%3A18859))

<img width="378" alt="Screenshot 2020-09-10 at 14 34 28" src="https://user-images.githubusercontent.com/20326070/92729656-f1fe0800-f372-11ea-91f7-10ef50d3a31f.png">
<img width="377" alt="Screenshot 2020-09-10 at 14 34 35" src="https://user-images.githubusercontent.com/20326070/92729662-f4606200-f372-11ea-91e6-afe81862c2e5.png">

---

- The 'Code editors' category is missing (open-in-editor can be already added there)

---

- when the 'Code analysis' filter is enabled, there are no extensions in the category. We should keep the same list as currently:
<img width="1139" alt="Screenshot 2020-09-10 at 14 38 42" src="https://user-images.githubusercontent.com/20326070/92729926-5b7e1680-f373-11ea-8fad-fddd308f380f.png">

---

- focus on the category filter stays there after turning it on and off. It's a similar behavior as there was previously in the toggle (before you fixed it with your last PR). 

<img width="131" alt="Screenshot 2020-09-10 at 14 39 30" src="https://user-images.githubusercontent.com/20326070/92730101-94b68680-f373-11ea-92f8-5de5d42c444e.png">
<img width="135" alt="Screenshot 2020-09-10 at 14 39 25" src="https://user-images.githubusercontent.com/20326070/92730105-95e7b380-f373-11ea-8f1d-818bced5ff96.png">

---

- the last breadcrumb on the extension page should be muted

<img width="138" alt="Screenshot 2020-09-10 at 14 41 11" src="https://user-images.githubusercontent.com/20326070/92730212-bc0d5380-f373-11ea-89db-d530ca8c5b46.png">

AlicjaSuska commented:
Related Issues:

Some extensions need a dark mode variant #13666 

and 

Add custom confirmation popup when enabling the extension #13664

Would it be possible to address those as well @tjkandala ?

tjkandala commented:
TODO: Consider effect of [React 17 changes](https://reactjs.org/blog/2020/08/10/react-v17-rc.html#potential-issues) for `useTimeoutManager`

davejrt commented:
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.20 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

tjkandala commented:
> * 'Other' should be the last category in the filters

Changed the order and made sure that programming languages are always rendered last so users aren't confused when they click 'show more extensions'.

> * CTA banner body text should have regular body color, this one is very light (light theme only)

Updated `web-content.scss` to set body color on light theme as well.

> * focus on the category filter stays there after turning it on and off. It's a similar behavior as there was previously in the toggle (before you fixed it with your last PR).

This is a bootstrap problem unfortunately. It seems that there's interest among Bootstrap developers to adopt a focus-visible polyfill, so it may change soon.

> * the last breadcrumb on the extension page should be muted

Changed in #13772

tjkandala commented:
@felixfbecker There's a good bit of new code I'd like review on, particularly with the extension card animation and toggle modal

felixfbecker commented on .vscode/settings.json:
```diff
@@ -40,7 +40,7 @@
   "typescript.format.semicolons": "remove",
   "typescript.tsc.autoDetect": "off",
-  "typescript.tsdk": "node_modules/typescript/lib",
+  "typescript.tsdk": "/home/tj/sourcegraph/node_modules/typescript/lib",
```
```suggestion
  "typescript.tsdk": "node_modules/typescript/lib",
```

felixfbecker commented on shared/src/components/ToggleBig.test.tsx:
```diff
@@ -0,0 +1,25 @@
+        const component = mount(<ToggleBig onToggle={onToggle} disabled={true} />)
+
+        component.find('.toggle-big').simulate('click')
+        sinon.assert.notCalled(onToggle)
```
👍

felixfbecker commented on shared/src/util/useTimeoutManager.ts:
```diff
@@ -0,0 +1,50 @@
+
+    // eslint-disable-next-line arrow-body-style
+    useEffect(() => {
+        /**
+         * https://reactjs.org/blog/2020/08/10/react-v17-rc.html#potential-issues
+         *
+         */
```
**nitpick**: this seems a bit excessive
```suggestion
        // https://reactjs.org/blog/2020/08/10/react-v17-rc.html#potential-issues
```

felixfbecker commented on shared/src/util/useTimeoutManager.ts:
```diff
@@ -0,0 +1,50 @@
+export function useTimeoutManager(): TimeoutManager {
+    const timeoutIDReference = useRef<{ timeoutID: number | undefined }>({ timeoutID: undefined })
+
+    // eslint-disable-next-line arrow-body-style
```
**question**: are you sure there's a violation here?
```suggestion
```

tjkandala replied:
Good catch, this was an issue before I made it React 17-proof (looked like `useEffect(() => () => clearTimeout(timeoutIDReference.current))`)

felixfbecker commented on web/src/components/ModalContainer.tsx:
```diff
@@ -1,24 +1,85 @@
+                focusedElement &&
+                (focusedElement instanceof HTMLButtonElement ||
+                    focusedElement instanceof HTMLAnchorElement ||
+                    focusedElement instanceof HTMLDivElement)
```
**question**: I'm confused by this check, it looks like it checks whether the element is an interactive element, but why the check for `<div>`? (and why does it have to check at all?)

tjkandala replied:
I'll simplify the check to just HTML or SVG Elements ([DOM type definitions](https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L8000)). `Element` doesn't have the `focus` method, but it looks like only `HTMLElement` or `SVGElement` extend `Element`, and they both have `focus`. Should I cast to `HTMLOrSVGElement` or check `instanceof`?

felixfbecker replied:
Ah, that makes sense. I would do `instanceof` since it's easier

felixfbecker commented on web/src/components/ModalContainer.tsx:
```diff
@@ -1,24 +1,85 @@
-                        <span onClick={this.props.onClose}>
+export const ModalContainer: React.FunctionComponent<Props> = ({ onClose, hideCloseIcon, className, children }) => {
+    const containerReference = useRef<HTMLDivElement>(null)
+    const bodyReference = useRef<HTMLElement>(null)
```
**suggestion**: when I first read this, it took me a while to understand because I thought this meant a reference to the `<body>`. Perhaps this could be renamed to `modalBodyReference`?
```suggestion
    const modalBodyReference = useRef<HTMLElement>(null)
```

felixfbecker commented on web/src/components/ModalContainer.tsx:
```diff
@@ -1,24 +1,85 @@
+                onClose()
+            }
+        }
+        // TODO(tj): Don't close modal when user mouse down inside body then mouse up outside of div?
+        document.addEventListener('mousedown', handleMouseDownOutside)
+
+        return () => document.removeEventListener('mousedown', handleMouseDownOutside)
```
**suggestion**: wouldn't that issue be solved by listening for `click`?
```suggestion
        document.addEventListener('click', handleMouseDownOutside)
        return () => document.removeEventListener('click', handleMouseDownOutside)
```

tjkandala replied:
I left that comment when I was listening for `click`, actually. `mousedown` solves that, but now users don't have a chance to course-correct and move their mouse over the modal. My new solution involves coordinating mouseup and mousedown. The intention is that the modal only closes when the user both presses and releases outside of the modal

felixfbecker commented on web/src/components/ModalContainer.tsx:
```diff
@@ -1,24 +1,85 @@
+    // Close modal when user presses `esc` key
+    const onKeyDown: React.KeyboardEventHandler<HTMLDivElement> = useCallback(
+        event => {
+            switch (event.key) {
+                case Key.Escape: {
+                    if (onClose) {
+                        onClose()
+                    }
+                }
+            }
```
```suggestion
            if (event.key === Key.Escape) {
                onClose?.()
            }
```

felixfbecker commented on web/src/components/ModalContainer.tsx:
```diff
@@ -1,24 +1,85 @@
+    return (
+        <div
+            ref={containerReference}
+            tabIndex={-1}
```
**question**: `div`s are not naturally focusable, does this serve a purpose? Like some effect on children?

tjkandala replied:
Yeah, so the user starts at the body, which might not be directly after the button they clicked on to open the modal. I noticed this behavior with `<SavedSearchModal />`, where the modal is rendered before the "Save this search query" button in the DOM tree. I'm about to implement focus trapping, so I'll focus on the first naturally focusable element instead.

felixfbecker commented on web/src/extensions/ExtensionBanner.scss:
```diff
@@ -0,0 +1,6 @@
+.extension-banner {
+    &__icon {
+        height: calc(4rem - 2px);
+        width: calc(4.5rem - 3px) '';
```
```suggestion
        width: calc(4.5rem - 3px);
```

felixfbecker commented on web/src/extensions/ExtensionBanner.story.tsx:
```diff
@@ -0,0 +1,13 @@
+
+const { add } = storiesOf('web/Extensions', module).addDecorator(story => <div className="p-4">{story()}</div>)
+
+add('Extension banner cta', () => <WebStory>{() => <ExtensionBanner />}</WebStory>, {
```
**suggestion**: name by component name
```suggestion
add('ExtensionBanner', () => <WebStory>{() => <ExtensionBanner />}</WebStory>, {
```

felixfbecker commented on web/src/extensions/extension/ExtensionAreaHeader.scss:
```diff
@@ -1,10 +1,40 @@
+    @include media-breakpoint-up(md) {
+        &__wrapper {
+            display: grid;
+            grid-template-columns: auto 120px 120px;
```
```suggestion
            grid-template-columns: auto 7.5rem 7.5rem;
```

felixfbecker commented on web/src/extensions/extension/ExtensionAreaHeader.tsx:
```diff
@@ -141,3 +162,24 @@ export const ExtensionAreaHeader: React.FunctionComponent<ExtensionAreaHeaderPro
+                                        </small>
+                                    </div>
+                                )} */
+}
```
Commented out chunk?

felixfbecker commented on web/src/extensions/icons.tsx:
```diff
@@ -0,0 +1,189 @@
+
+/** Default icon for Sourcegraph extensions displayed when enabled (visual feedback) */
+export const DefaultIconEnabled: React.FunctionComponent<IconProps & ThemeProps> = ({ className, isLightTheme }) =>
+    isLightTheme ? (
```
**thought**: would be cool if the SVG adapted to the theme with CSS only

tjkandala replied:
backlog investigation :)

felixfbecker commented on web/src/extensions/icons.tsx:
```diff
@@ -0,0 +1,189 @@
+)
+
+export const extensionBannerIconURL =
+    'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAM8AAAC4CAYAAABEprgmAAAACXBIWXMAACE4AAAhOAFFljFgAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAFx7SURBVHgB7b1tsG7JVR7Wu7v3e869dzSj+ZQGIYRsS5YTDNhWDFbwD1XiKsqVhKpUZSoO2MTByTgBJoaiKn/vr5SrbENZWIknUBADhtT9kR+xK1UpXDXFlxAqibIDhZGQQYCMRprRzGju1znv7t3tvT6e1Wu/7xlZQnPnHp339My57969+7vXV69evXoIlyH8D89+dHz3m24+dC0+8NapnD6QQn5wSMNY53kTwhxi2mznqdyNOW2PYnrlNLYX/9XL ...
```
**nitpick**: embedding a PNG (binary by nature) in Base64 (even larger) in our source code does not seem right. Why does this have to be a PNG, instead of an SVG?

tjkandala replied:
![image](https://user-images.githubusercontent.com/37420160/94074179-c04f5d00-fdc6-11ea-81f0-40942c05d1ef.png)
![image](https://user-images.githubusercontent.com/37420160/94074279-f68cdc80-fdc6-11ea-9e9e-574e307a6481.png)

When I used an SVG, the opacity would be lower at the bottom of the the screen than at the top. Do you have any guesses as to why that happens? @marekweb narrowed it down to `<linearGradient />`; this also happened with other icons that used `<linearGradient />`, but not with icons that didn't. As for Base64, I did that to prevent reflow when loading the png.

felixfbecker replied:
Sounds like the linear gradient is somehow anchored to the viewport instead of to the SVG viewBox 🤔 
Does it happen in all browsers? Does it also happen when embedding the SVG with `<img>`? We can put files in our `assets` folder too.

tjkandala replied:
The reflow happened when the PNG was in the assets folder. I'll revert to SVG and test on different browsers after fixing the blocking issues

tjkandala replied:
backlog

felixfbecker commented on web/src/extensions/ExtensionCard.tsx:
```diff
@@ -32,96 +35,189 @@ const stopPropagation: React.MouseEventHandler<HTMLElement> = event => {
+    const [change, setChange] = React.useState<'enabled' | 'disabled' | null>(null)
+    const feedbackManager = useTimeoutManager()
+
+    // Add class that triggers box shadow animation .3s after enabled, and remove it 1s later
+    const [showShadow, setShowShadow] = React.useState(false)
+    const startAnimationManager = useTimeoutManager()
+    const endAnimationManager = useTimeoutManager()
+
+    const onToggleChange = React.useCallback(
+        (enabled: boolean): void => {
+            if (enabled) {
+                setChange('enabled')
+                // not using transition-delay so the shadow will immediately disappear on disable
+                startAnimationManager.setTimeout(() => {
+                    setShowShadow(true)
+                }, 300)
+                endAnimationManager.setTimeout(() => {
+                    setShowShadow(false)
+                }, 1300)
+            } else {
+                setChange('disabled')
+                setShowShadow(false)
+                startAnimationManager.cancelTimeout()
+                endAnimationManager.cancelTimeout()
+            }
+            feedbackManager.setTimeout(() => {
+                setChange(null)
+            }, FEEDBACK_DELAY)
+        },
+        [feedbackManager, startAnimationManager, endAnimationManager]
+    )
```
**question**: did you consider doing the animation with CSS? I.e. adding a class, but doing the rest with `transition` or `@keyframes`. (this is not an issue as long as it's performant, just curious if there was anything blocking a CSS approach)

tjkandala replied:
I believed that handling cancellation would be easier with this approach of just toggling classes and adding/removing with timeouts.

felixfbecker replied:
Makes sense! I'm actually not sure how to cancel CSS animations, but maybe when the class that applies the animation is removed, the animation is cancelled too. Again this was just curiosity though, I would not recommend spending time on this unless everything else is ready and you really want to :)

felixfbecker commented on web/src/extensions/ExtensionToggle.tsx:
```diff
@@ -1,127 +1,128 @@
-    public componentWillUnmount(): void {
-        this.subscriptions.unsubscribe()
-    }
+            // eslint-disable-next-line no-void
+            void platformContext.updateSettings(highestPrecedenceSubject.subject.id, {
```
**issue**: this disables the ESLint rule to avoid handling the error, but shouldn't the error be handled? Especially since there is an optimistic update, it seems that if the persistence fails, the optimistic update should be rolled back and the error displayed somewhere

felixfbecker reviewed (commented):
Wow, this has grown into a pretty big PR. Kudos for all the work that went into this. Would like your eyes too @marekweb since it's easy to miss something at this size. I'll also do some manual testing tomorrow.
Note also that the build is failing and there are currently merge conflicts so it would be good to merge in `main`.

felixfbecker commented:
@tjkandala I'd propose to first focus on anything blocking merge (merge conflicts, CI, left-over commented-out code, anything marked **issue** etc) and punt on anything not blocking merge (using CSS `@keyframes`, scope trapping (?), ...) just to make sure we eliminate any risk of this PR not being merged by EOW :)

tjkandala commented:
> @tjkandala I'd propose to first focus on anything blocking merge (merge conflicts, CI, left-over commented-out code, anything marked **issue** etc)

Will do!

> and punt on anything not blocking merge (using CSS `@keyframes`

+1

> scope trapping (?), ...) 

Users can tab out of the modal to "underneath" it, so we should implement a way to loop around to the first focusable element from the last focusable element. I'll make sure to work on this _after_ cleaning up merging :).

tjkandala commented on web/src/extensions/ExtensionPermissionModal.tsx:
```diff
@@ -0,0 +1,35 @@
+            component={
+                <div className="extension-permission-modal p-4">
+                    <h3>Add {extensionName || extensionID} Sourcegraph extension?</h3>
+                    <p className="mb-0 mt-3">It will be able to:</p>
+                    <p className="m-0">- read repositories and files you view using Sourcegraph</p>
+                    <p className="m-0">- read and change your Sourcegraph settings</p>
```
![Screenshot from 2020-09-23 18-53-25](https://user-images.githubusercontent.com/37420160/94082094-225f9080-fdce-11ea-98d3-64a788f53c0a.png)

What should we do about the `padding-inline-start`? cc @AlicjaSuska

tjkandala replied:
I don't think it looks bad fwiw

tjkandala commented:
<details open>
<summary>Error after enabling extension</summary>
<img src="https://user-images.githubusercontent.com/37420160/94092957-04078e00-fdea-11ea-996d-509e7eb08ac9.gif" />
</details>

<details>
<summary>Error after toggling multiple times in a row</summary>
<img src="https://user-images.githubusercontent.com/37420160/94092959-0538bb00-fdea-11ea-9f6b-76290c7bfc68.gif" />
</details>

Optimistic update error handling. I scripted an error to throw 900ms after `platformContext.updateSettings` calls.
The new implementation may need a small review :).

tjkandala commented on web/src/extensions/ExtensionToggle.tsx:
```diff
@@ -1,127 +1,193 @@
-            />
-        )
-    }
+    const optimisticRollbackPipeline = useCallback(
```
**self-nit**: this shouldn't be called `optimisticRollbackPipeline` if it's being passed to `createOptimisticRollbackPipeline`. This should be named `onOptimisticError` or something similar.

tjkandala commented on web/src/extensions/ExtensionToggle.tsx:
```diff
@@ -1,127 +1,193 @@
-                                value: enabled,
-                            })
-                        )
+function createOptimisticRollbackPipeline<T>(
```
**thought**: should probably write a hook that encapsulates this, `useEventObservable`, and `useState` into one `useOptimisticState` hook. Will probably take initial state and on error callback, and return `nextOptimisticState`, which takes the same type as now but sets state with `optimisticValue`.  (after merging ofc)

felixfbecker commented:
Caught some small things in manual testing:

- [ ] The (?) icon here is not linked, I tried clicking on it while testing (it's linked in Figma). ![image](https://user-images.githubusercontent.com/10532611/94131446-35ba3c80-fe5e-11ea-86fa-880ce7869c48.png)
- [ ] The `shadow` class on the input overrides the focus ring, which also uses `box-shadow`. I think this can be solved easily by wrapping the input in a `div` that has the `shadow` class. 
![image](https://user-images.githubusercontent.com/10532611/94134133-e37b1a80-fe61-11ea-9ca1-267015977146.png)
- [ ] The flash message on the registry page says "off", but on the extension page it says "disabled". I think we should go with "disabled" consistently 
![image](https://user-images.githubusercontent.com/10532611/94134257-11605f00-fe62-11ea-817c-5f7e474542b6.png)
- [ ] What do you think about making the green highlight box-shadow animation like 2x faster to make it feel snazzier?

None of these necessarily have to block merging.

tjkandala commented:
>   ![image](https://user-images.githubusercontent.com/10532611/94134257-11605f00-fe62-11ea-817c-5f7e474542b6.png)
> * [ ]  The breadcrumb for an extension on a private registry shows the registry as the publisher and the publisher as the name, not showing the name

This is fixed locally (implemented splitExtensionIDs function to handle private registry extensions)

>   ![image](https://user-images.githubusercontent.com/10532611/94134594-83d13f00-fe62-11ea-88c6-310a0649d30d.png)
> * [ ]  The toggle on extension pages looks a bit oddly positioned, it's not really aligned with anything. I think I would have expected it to be right-aligned inside the container? @AlicjaSuska  What's the intention here, I saw in the design it was also not right-aligned?

This will be centered vertically, but I think it should be right-aligned.

>   ![image](https://user-images.githubusercontent.com/10532611/94163064-b2addc00-fe87-11ea-93f5-29b9d9bd120f.png)
>   ![image](https://user-images.githubusercontent.com/10532611/94163272-edb00f80-fe87-11ea-8894-3c3f60f5512b.png)
> * [ ]  What do you think about making the green highlight box-shadow animation like 2x faster to make it feel snazzier?

I'll try it!

tjkandala commented:
Update: Finished all blocking and non-blocking issues that I'm aware of, other than the dark theme text-muted issue that I asked about in Slack.

<details open>
<summary>Sourcegraph logo</summary>
<img src="https://user-images.githubusercontent.com/37420160/94221843-eace1280-feb9-11ea-9b41-a05aa70f6ca8.png" />
</details>

<details open>
<summary>New category filter logic</summary>
<img src="https://user-images.githubusercontent.com/37420160/94222307-00900780-febb-11ea-8cda-8cdf9cdfdd6e.gif" />
</details>


- Quicker animation
- Vertically centered toggle on extension details page
- btn-sm extension area header tabs

@AlicjaSuska @felixfbecker any other concerns before merging?

marekweb commented:
It looks good to me. It's a remarkable amount of work and I'm excited that it's landing 💯 🎉

felixfbecker commented:
@AlicjaSuska could you review the UI in Storybook? 🙌

AlicjaSuska commented:
Thank you for making all the changes @tjkandala Here are some points from me:

**Extension card**
- in the extension card 'sourcegraph' should be bolded out. 'by **sourcegraph**' this is the case only for Sourcegraph - authored extensions. Other publishers' names should stay regular
- SG logo is too big - the bounding box should be 12px, not our standard 20px

**Extension details page**
Implementation corresponds with the initial design. Anyways, according to the latest discussions with @sqs we will be changing the approach (I apologise it's causing more work to do here ☹️ ):
- breadcrumbs will be removed from the web-oriented pages. 'Partial' breadcrumbs will be on some deeper nested pages (please, see the Rob's proposition in this [doc](https://docs.google.com/document/d/1XwuUg1gV4LA_GaXvjguCLLscDDt2DRQACBa_SDS_ICE/edit?usp=sharing))
- header will come back to publisher/author structure

I'm not sure if we want to wait for those changes or just release with old breadcrumbs and page title structure (publisher/name)? @tjkandala @felixfbecker wdyt?

Fixes not related to the items above:
- tabs shouldn't have a blue link color, it should stay as it is now
<img width="220" alt="Screenshot 2020-09-25 at 15 51 50" src="https://user-images.githubusercontent.com/20326070/94283613-852d6500-ff51-11ea-8cac-d34ba4091242.png">

- tags and categories should have a outline button small styles, no blue link text
<img width="205" alt="Screenshot 2020-09-25 at 15 51 54" src="https://user-images.githubusercontent.com/20326070/94283627-89f21900-ff51-11ea-9fb0-1a1177e80771.png">

AlicjaSuska commented:
I've added comments to Chromatic as well

AlicjaSuska commented:
@tjkandala I've seen you resolved my comments in Chromatic but there are no changes made. Can you provide more details? Thank you!
